authorization/v1/types.go

```diff
@@ -460,12 +460,15 @@ type RoleBindingRestriction struct {
 // field must be non-nil.
 type RoleBindingRestrictionSpec struct {
 	// UserRestriction matches against user subjects.
+	// +nullable
```

Comment:
What's the difference between this and optional?  Both?

Reply:
`+nullable` has an effect, which is to add `nullable: true` to the fields in the openapi validation

`+optional` actually currently doesn't do anything. Upstream controller-tools only checks for it to remove it from descriptions in the openapi spec, and instead uses `omitempty` tags to infer if fields are really optional or required. (We forked controller-tools to invert their default behavior of assuming anything not optional was required). In some places we're still keeping it around for info purposes but it's not required